### PR TITLE
stress-ng 0.09.04 (new formula)

### PR DESCRIPTION
**Pull request by jacobbednarz**
Adds a new formula for [`stress-ng`](https://github.com/ColinIanKing/stress-ng) which is useful for generating
system load and testing of various components like CPU, memory,
syscalls.

--- 

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/stress-ng.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class StressNg < Formula
+  desc "Stress test a computer system in various selectable ways"
+  homepage "http://kernel.ubuntu.com/~cking/stress-ng/"
+  url "https://github.com/ColinIanKing/stress-ng/archive/V0.09.04.tar.gz"
```
Comment (fxcoudert) [inline Formula/stress-ng.rb]: Let's use the official release tarball: http://kernel.ubuntu.com/~cking/tarballs/stress-ng/stress-ng-0.09.06.tar.xz

File: Formula/stress-ng.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  sha256 "83bb1f11e66df753f16dd924c36fea9df20281ca85d49d07c15d970fe902baa4"
+
+  def install
+    system "make"
+    bin.install "stress-ng"
```
Comment (fxcoudert) [inline Formula/stress-ng.rb]: Let's use `make install`, but before that we need to replace `/usr` with `#{prefix}` in the `Makefile`.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/stress-ng.rb (hunk 1 above)
Comment (jacobbednarz) [reply Formula/stress-ng.rb]: Fixed in 1f78751

File: Formula/stress-ng.rb (hunk 2 above)
Comment (jacobbednarz) [reply Formula/stress-ng.rb]: Thanks! I updated this in 2baa869

**Comment by jacobbednarz** [comment]
This is still failing on El Capitan

```
09:00:26 Undefined symbols for architecture x86_64:
09:00:26   "_getentropy", referenced from:
09:00:26       _shim_getrandom in shim.o
09:00:26 ld: symbol(s) not found for architecture x86_64
09:00:26 clang: error: linker command failed with exit code 1 (use -v to see invocation)
09:00:26 make[1]: *** [stress-ng] Error 1
09:00:26 make: *** [all_config] Error 2
09:00:26 
```

I don't know what I could be missing here :/ Do you have any thoughts? Perhaps we restrict this to installing on Sierra + High Sierra?

**Comment by fxcoudert** [comment]
`depends_on :macos => :sierra # needs getentropy` is fine by me

**Review by fxcoudert — APPROVED** [review]

File: Formula/stress-ng.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  depends_on :macos => :sierra
+
+  def install
+    inreplace "Makefile", "/usr", prefix
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/stress-ng", "-c 1", "-t 1"
```
Comment (ilovezfs) [inline Formula/stress-ng.rb]: Can we have this verify something beyond just a zero exit code?

**Review by ilovezfs — DISMISSED** [review]

File: Formula/stress-ng.rb (hunk 3 above)
Comment (jacobbednarz) [reply Formula/stress-ng.rb]: We could look to assert the output matches. Would that be sufficient for what you're after here?

Comment (jacobbednarz) [reply Formula/stress-ng.rb]: Addressed in 8eae3e3

**Comment by jacobbednarz** [comment]
Thanks for the feedback @ilovezfs and @fxcoudert. This has now been updated and should be ready for your review.

**Comment by fxcoudert** [comment]
Congratulations @jacobbednarz on your first contribution to Homebrew! 🎆

**Pull request closed without merging**